### DISCUSSION TITLE
URL "encoded" password handling?

### DISCUSSION THREAD

**Discussion by fhuse** [sqlalchemy/sqlalchemy#10144] [Usage Questions]
When a connection URL contains "%[0-F][0-F]" sequence. It will be replaced by the ASCII character by the given the hex code.

user: TESTUSER
password: %42%41%44%4254F%59

eng=create_engine("oracle://TESTUSER:%42%41%44%4254F%59@TESTSERVER/")
eng.connect()

Will fail becuase during the URL parsing the '%42%41%44%4254F%59' will be replaced with 'BADBOY'.

**Comment by CaselIT** [comment]
Hi,

As documented you need to url encode the user and password if they have special characters
